bioimageio/spec/commands.py

```diff
@@ -19,9 +22,26 @@ def validate(
     if update_format_inner is None:
         update_format_inner = update_format
 
-    source_name = str(rdf_source.get("name") if isinstance(rdf_source, dict) else rdf_source)
+    if isinstance(rdf_source, (BytesIO, StringIO)):
+        rdf_source = rdf_source.read()
+    elif isinstance(rdf_source, os.PathLike):
+        # zipfile.is_zipfile cannot deal with all os.PathLike objects (e.g. pytest's `LocalPath`),
+        # so we cast to pathlib.Path
+        rdf_source = pathlib.Path(rdf_source)
+
+    source_name = str(
```

Comment:
Why are we doing this before loading the rdf as a dict in case it's online?

Reply:
`source_name` is only used for more concise error message